docs/learn/migrate/solana.mdx

```diff
@@ -1,8 +1,718 @@
+        assert_eq!(retrieved_data, Some(user_data));
+
+        // Update balance
+        let new_balance = 200;
+        let updated_data = client.update_balance(&user, &new_balance);
+        assert_eq!(updated_data.balance, new_balance);
+    }
+}
+```
+
+## Deployment
+
+Once your contract is ready, here's how to deploy it:
+
+### Local Development
+
+```bash
+# Start local Stellar network using Stellar CLI
+stellar container start local
+
+# Build your contract
+cargo build --target wasm32-unknown-unknown --release
+
+# Deploy to local network
+stellar contract deploy \
+  --wasm target/wasm32-unknown-unknown/release/my_soroban_contract.wasm \
+  --source-account your_account_name \
+  --network local \
+  --alias my_contract
+```
+
+### Testnet Deployment
+
+```bash
+# Configure for testnet
+stellar network add testnet
+
+# Deploy to testnet
+stellar contract deploy \
+  --wasm target/wasm32-unknown-unknown/release/my_soroban_contract.wasm \
+  --source-account your_account_name \
+  --network testnet \
+  --alias my_contract
+```
+
+### Mainnet Deployment
+
+```bash
+# Configure for mainnet
+stellar network add mainnet
+
+# Deploy to mainnet
+stellar contract deploy \
+  --wasm target/wasm32-unknown-unknown/release/my_soroban_contract.wasm \
+  --source-account your_account_name \
+  --network mainnet \
+  --alias my_contract
+```
+
+## Common Challenges and Solutions
+
+### 1. Authorization Patterns
+
+**Challenge**: Understanding when and how to use `require_auth()`
+
+**Solution**: Always call `require_auth()` on the user address for any function that modifies state or performs actions on behalf of a user.
+
+```rust
+pub fn update_user_data(env: Env, user: Address, new_data: String) -> Result<(), CustomError> {
+    // This ensures only the user can call this function
+    user.require_auth();
+
+    // Now safe to update user data
+    env.storage().instance().set(&user, &new_data);
+    Ok(())
+}
+```
+
+### 2. Storage Patterns
+
+**Challenge**: Moving from Solana's account-based storage to Stellar's key-value storage
+
+**Solution**: Use Stellar's storage abstraction with proper key management:
+
+```rust
+// Store data with user address as key
+env.storage().instance().set(&user, &user_data);
+
+// Retrieve data
+let user_data: Option<UserData> = env.storage().instance().get(&user);
+```
+
+### 3. Cross-Contract Calls
+
+**Challenge**: Converting from Solana CPI to Stellar contract calls
+
+**Solution**: Use the `contractimport!` macro for type-safe interactions:
+
+```rust
+contractimport!(
+    file = "../token_contract/target/wasm32-unknown-unknown/release/token_contract.wasm"
+);
+
+// Now you have a typed client
+let token_client = TokenContractClient::new(&env, &token_address);
+token_client.transfer(&from, &to, &amount);
+```
+
+## Solana vs. Soroban: Key Differences Summary
+
+| Aspect | Solana | Soroban (Stellar) |
+| --- | --- | --- |
+| **Virtual Machine** | BPF (Berkeley Packet Filter) | WebAssembly (WASM) |
```

Comment:
nit: Please use `Wasm` instead of `WASM` everywhere, as this is the official abbreviation for WebAseembly